btk/measure.py

```diff
@@ -1,175 +1,239 @@
-        self.segmentation = None
-
-    def get_centers(self, image):
-        """Return centers detected when object detection and photometry
-        is done on input image with SEP.
-        It also initializes the self.catalog and self.segmentation attributes
-        of the class object.
-        Args:
-            image: Image (single band) of galaxy to perform measurement on.
-        Returns:
-            centers: x and y coordinates of detected  centroids
-        """
-        bkg = sep.Background(image)
-        self.catalog, self.segmentation = sep.extract(
-            image, 1.5, err=bkg.globalrms, segmentation_map=True
-        )
-        centers = np.stack((self.catalog["x"], self.catalog["y"]), axis=1)
-        return centers
+    Attributes:
+        self.measure_functions (list): List of functions that take as input the output from
+                                        DrawBlendsGenerator and return the output of a measurement
+                                        (see module docstring).
+    """
 
-    def get_deblended_images(self, data, index):
-        """Performs SEP detection on the band-coadd image and returns the
-        detected peaks.
-        Args:
-            data (dict): Output generated by btk.draw_blends containing blended
-                images, isolated images, observing conditions and blend
-                catalog, for a given batch.
-            index (int): Index number of blend scene in the batch to preform
-                measurement on.
-        Returns:
-            dict with the centers of sources detected by SEP detection
-            algorithm.
-        """
-        image = np.mean(data["blend_images"][index], axis=0)
-        peaks = self.get_centers(image)
-        return {"deblend_image": None, "peaks": peaks}
+    measure_params = {"deblended_images", "catalog", "segmentation"}
 
-
-class MeasureGenerator:
     def __init__(
         self,
-        measurement_params,
+        measure_functions,
         draw_blend_generator,
         multiprocessing=False,
         cpus=1,
         verbose=False,
     ):
-        """Generates output of deblender and measurement algorithm.
+        """
 
         Args:
-            measurement_params: Instance from class
-                                `btk.measure.Measurement_params`.
+            measure_function: Function or list of functions that returns a dict with
+                              measurements given output from the draw_blend_generator.
             draw_blend_generator: Generator that outputs dict with blended images,
-                                  isolated images, observing conditions and blend
-                                  catalog.
+                                  isolated images, blend catalog, wcs info, and psf.
             multiprocessing: If true performs multiprocessing of measurement.
             cpus: If multiprocessing is True, then number of parallel processes to
                  run [Default :1].
```

Comment:
Not in this PR, but I think these two arguments are redundant.
```python
multiprocessing = True if cpu > 1 else False
```

Reply:
Thanks @aboucaud , I created a new issue to address this.